pkg/yang/types.go

```diff
@@ -350,22 +356,40 @@ check:
 	for _, p := range y.Pattern {
 		patterns[p] = true
 	}
-	for _, pv := range t.Pattern {
-		p := pv.Name
+
+	checkAndAddPattern := func(n Node, p string) {
```

Comment:
I wonder whether this is the right place to do this parsing -- essentially, it means that downstream consumers never know about the extension, which I'm not sure is the right thing. Particularly, let's assume that there's some arbitrary tool that today does implement w3c patterns based on the output of goyang, in this model - if they specify `UsePosixPattern` because one module that they're using uses this form of pattern, then they will potentially get an error because there's no signal as to whether it was the POSIX pattern, or the regular pattern statement that was used to populate `(yang.Entry).Pattern`.

It seems like an alternate design would be to have a `POSIXPattern` field in the `Entry`, and then allow downstream tooling to determine which it uses.

Reply:
Hmm I like that very much. It means the conversion has the steps of
1. update goyang, which doesn't break anything.
2. update models & downstream code.

I like it.